app/src/main/java/com/mobileasone/dagger2workshop/di/DetailNoteFragmentModule.kt

```diff
@@ -0,0 +1,17 @@
+package com.mobileasone.dagger2workshop.di
+
+import com.mobileasone.dagger2workshop.presentation.notedetail.DetailNoteFragmentPresenter
+import com.mobileasone.dagger2workshop.presentation.notedetail.DetailNoteFragmentPresenterImpl
+import dagger.Module
+import dagger.Provides
+
+/**
+ * This module will provide dependencies for DetailNoteFragment
+ */
+@Module
+class DetailNoteFragmentModule {
+
+    @Provides
+    fun noteListFragmentPresenter(presenter: DetailNoteFragmentPresenterImpl): DetailNoteFragmentPresenter = presenter
```

Comment:
Another way to provide instances of objects.
To provide instances of DetailNoteFragmentPresenter we say how to create it using @Inject annotation in the constructor of [DetailNoteFragmentPresenterImpl.kt](https://github.com/Jamargle/Dagger2Workshop/pull/6/files#diff-aa9d20461ac1116401916d06ddb1ec2f).